### PR TITLE
Allow loading lookup data in Oak Functions loader

### PR DESCRIPTION
**Pull request by tiziano88**
Not actually used in the server implementation yet.

Ref #1930

# Checklist

- [ ] Pull request affects core Oak functionality (e.g. runtime, SDK, ABI)
  - [ ] I have written tests that cover the code changes.
  - [ ] I have checked that these tests are run by
        [Cloudbuild](/cloudbuild.yaml)
  - [ ] I have updated [documentation](/docs/) accordingly.
  - [ ] I have raised an [issue](https://github.com/project-oak/oak/issues) to
        cover any TODOs and/or unfinished work.
- [ ] Pull request includes prototype/experimental work that is under
      construction.

### CODE REVIEW

File: oak_functions/loader/src/main.rs [hunk 1]
```diff
@@ -43,6 +48,87 @@ pub struct Opt {
...
+            .expect("could not lock entries for read")
+            .get(key)
+            .cloned()
+    }
+}
+
+fn load_lookup_entries<B: bytes::Buf>(
+    lookup_data_buffer: B,
+) -> anyhow::Result<HashMap<Vec<u8>, Vec<u8>>> {
+    let mut lookup_data_buffer = lookup_data_buffer;
```
Comment (conradgrobler) [inline oak_functions/loader/src/main.rs:124]: Why is this needed?

**Review by conradgrobler — APPROVED** [review]

**Review by rbehjati — APPROVED** [review]
LGTM!

File: oak_functions/loader/src/main.rs (hunk 1 above)
Comment (tiziano88) [reply oak_functions/loader/src/main.rs:132]: It's just a rebinding to allow for mutable methods to be invoked on it. The variable is passed by value already, so this is just a detail of the implementation of this method, it's not making things "more" mutable, since the value is already owned by this function.
